### PR TITLE
Improve error handling in git-ai-commit and add comprehensive tests

### PR DESCRIPTION
**Pull request by greenc-FNAL**
Primary goal: resolve CodeQL `py/empty-except` violations.

Additionally: refactor three functions to handle edge cases more robustly:

- **_kilo_auth_token**: Replace lenient `.get()` chaining with explicit
  `isinstance()` checks at each level (dict, provider, key string). Validates
  JSON structure before attempting field access, preventing silent failures on
  malformed credentials files.

- **_gh_cli_token**: Add explicit `return None` for non-zero exit codes and
  missing `gh` binary (FileNotFoundError). Previously relied on implicit
  falsy stdout check; now surfaces all failure modes uniformly.

- **_edit**: Wrap subprocess call in try-except to catch FileNotFoundError
  (missing editor) and CalledProcessError (non-zero exit). Raise _Error with
  descriptive messages instead of letting exceptions propagate. Ensures temp
  file cleanup via finally block even on error.

Finally, add comprehensive test suite
(`scripts/test/test_git_ai_commit.py`) covering all three functions with
28 test cases: malformed JSON, missing files, type mismatches,
subprocess failures, and happy paths. Tests use importlib to load the
script as a module and pytest fixtures for isolation.

### CODE REVIEW

**Comment by greenc-FNAL** [comment]
## ✅ 1 CodeQL alert resolved compared to main
- :white_check_mark: **Warning** [# 177](https://github.com/Framework-R-D/phlex/security/code-scanning/177) `py/empty-except` at `scripts/git-ai-commit:214:5` — 'except' clause does nothing but pass and there is no explanatory comment.

Review the [full CodeQL report](https://github.com/Framework-R-D/phlex/security/code-scanning) for details.

File: scripts/test/test_git_ai_commit.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    ) -> None:
+        """OSError while reading the file → None."""
+        p = tmp_path / "auth.json"
+        p.write_text("{}", encoding="utf-8")
+        p.chmod(0o000)
+        self._setup(monkeypatch, p)
+        try:
+            result = _M._kilo_auth_token()
+        finally:
+            p.chmod(0o644)
```
[the remark this answers is not in the document]
Comment (greenc-FNAL) [reply scripts/test/test_git_ai_commit.py]: Good catch. Replaced the `chmod(0o000)` approach with `patch("pathlib.Path.read_text", side_effect=OSError(...))`. This is deterministic regardless of whether tests run as root, and avoids leaving the file in an unreadable state if the assertion fails.

File: scripts/git-ai-commit [hunk 2]
```diff
@@ -527,7 +537,12 @@ def _edit(text: str) -> str:
...
         tmp = Path(fh.name)
     try:
-        subprocess.run([editor, str(tmp)], check=True)
+        try:
+            subprocess.run([editor, str(tmp)], check=True)
+        except FileNotFoundError:
+            raise _Error(f"Editor not found: {editor!r}. Set $EDITOR or $VISUAL.") from None
+        except subprocess.CalledProcessError as exc:
+            raise _Error(f"Editor exited with status {exc.returncode}.") from exc
         lines = tmp.read_text().splitlines()
```
[the remark this answers is not in the document]
Comment (greenc-FNAL) [reply scripts/git-ai-commit]: Fixed. `tmp.read_text()` is now wrapped in `try/except (OSError, UnicodeDecodeError)` which re-raises as `_Error("Could not read editor output: ..."))`. A corresponding test `test_read_error_after_edit_raises_error` has been added to `TestEdit` to cover this path.

**Review by knoepfel — APPROVED** [review]
